File: src-tauri/src/models/workspace_item.rs
use serde::{Deserialize, Serialize};

/// Workspace item entity
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorkspaceItem {
    pub id: String,
    #[serde(rename = "type")]
    pub item_type: WorkspaceItemType,
    pub original_path: String,
    pub current_path: String,
    pub thumbnail_path: Option<String>,
    pub title: String,
    pub created_at: String,
    pub updated_at: String,
    pub is_favorite: bool,
    pub metadata_json: Option<String>,
}

/// Type of workspace item
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum WorkspaceItemType {
    Image,
    Video,
}

impl std::fmt::Display for WorkspaceItemType {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            WorkspaceItemType::Image => write!(f, "image"),
            WorkspaceItemType::Video => write!(f, "video"),
        }
    }
}

File: src/AmeCapture.Domain/Entities/WorkspaceItem.cs
namespace AmeCapture.Domain.Entities;

public enum WorkspaceItemType
{
    Image,
    Video
}

public class WorkspaceItem
{
    public string Id { get; set; } = string.Empty;
    public WorkspaceItemType ItemType { get; set; }
    public string OriginalPath { get; set; } = string.Empty;
    public string CurrentPath { get; set; } = string.Empty;
    public string? ThumbnailPath { get; set; }
    public string Title { get; set; } = string.Empty;
    public string CreatedAt { get; set; } = string.Empty;
    public string UpdatedAt { get; set; } = string.Empty;
    public bool IsFavorite { get; set; }
    public string? MetadataJson { get; set; }
}
